bazel_ros2_rules/ros2/tools/dload_cc.bzl

```diff
@@ -30,8 +30,12 @@ using bazel::tools::cpp::runfiles::Runfiles;
 int main(int argc, const char * argv[]) {{
   std::string error;
   std::unique_ptr<Runfiles> runfiles(Runfiles::Create(argv[0], &error));
+  if (!runfiles && std::filesystem::is_symlink(argv[0])) {{
+    runfiles.reset(
+      Runfiles::Create(std::filesystem::read_symlink(argv[0]), &error));
+  }}
```

Comment:
@EricCousineau-TRI - this makes the test pass. Does it seem like the right fix to you?

It seems like the shim was unable to read its runfiles during `bazel run //:runfiles_direct_test` because `argv[0]` was a symlink. Resolving that symlink leads to the real executable adjacent to a `*.runfiles` directory and `*.runfiles_manifest` file.